apps/files/src/components/FileSharingSidebar.vue

```diff
@@ -294,25 +285,30 @@ export default {
     $_isCollaboratorShare (collaborator) {
       return userShareTypes.includes(collaborator.shareType)
     },
-    $_collaboratorsComparator (c1, c2) {
-      // resharer entries have displayName on first level,
-      // but shares/collaborators have it under the collaborator attribute
+    collaboratorsComparator (c1, c2) {
+      // Sorted by: type, direct, display name, creation date
       const c1DisplayName = c1.collaborator ? c1.collaborator.displayName : c1.displayName
       const c2DisplayName = c2.collaborator ? c2.collaborator.displayName : c2.displayName
       const name1 = c1DisplayName.toLowerCase().trim()
       const name2 = c2DisplayName.toLowerCase().trim()
-      // sorting priority 1: display name (lower case, ascending), 2: share type (groups first), 3: id (ascending)
-      if (name1 === name2) {
-        const c1GroupShare = c1.shareType === shareTypes.group
-        const c2GroupShare = c2.shareType === shareTypes.group
-        if (c1GroupShare === c2GroupShare) {
-          return textUtils.naturalSortCompare(c1.id + '', c2.id + '')
-        } else {
-          return c1GroupShare ? -1 : 1
+      const c1UserShare = c1.shareType === shareTypes.user || c1.shareType === shareTypes.remote
+      const c2UserShare = c2.shareType === shareTypes.user || c1.shareType === shareTypes.remote
+      const c1DirectShare = !c1.outgoing
```

Comment:
confusion here: with "outgoing" I'd understand "outgoing share" but not direct vs indirect

I wonder if it was related to the hack that used that attribute simply to find out whether the share entry comes from the sharestree vs the regular one. Maybe add a comment there to clarify.

I think I'll want to clean this up soon as "the hack" is starting to spread everywhere.
Or if you want to tackle it quickly, maybe we can simply add an attribute "indirect" additionally to "incoming" and "outgoing" to reduce confusion. Not optimal but would do.

Reply:
I originally added the indirect but than I saw there is the outgoing so I didn't want to introduce a new key for the same result. But I get there the confusion and didn't know that it was only a hack.

I'll adjust it.

Reply:
Thanks a lot and sorry for the old mess :-S

Reply:
Just went the quick way as you suggested and added there an `indirect` key. I'm not sure how you plan to tackle this in refactor so didn't want to touch it too much.